clients/src/main/java/org/apache/kafka/common/network/PlaintextChannelBuilder.java

```diff
@@ -34,6 +34,15 @@
 public class PlaintextChannelBuilder implements ChannelBuilder {
     private static final Logger log = LoggerFactory.getLogger(PlaintextChannelBuilder.class);
     private Map<String, ?> configs;
+    private final ListenerName listenerName;
+
+    /**
+     * Constructs a plaintext channel builder. ListenerName is provided only
+     * for server channel builder and will be null for client channel builder.
```

Comment:
It's actually a bit more complicated. It's non-null whenever it's instantiated in the broker and null otherwise. In the broker case, we create both server and client channel builders, but it's set in both (for sending and receiving requests respectively).